### PR TITLE
Upgrade the Nix build system

### PR DESCRIPTION
**Pull request by pepeiborra**
* Switch to the haskell-updates branch, so that we get Stackage nightly packages
* Remove the hard-coded list of packages, let cabal2nix find out
* A Github Action to exercise the Nix script and warm up a Cachix cache

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -701,6 +701,19 @@ $ cd install
...
+The instructions below show how to set up a Cachix binary cache and open a nix shell for local development.
+
+```shell
+$ cachix use haskell-language-server
+$ nix-shell
+$ cabal update
+$ cabal build
+```
+
+If you are looking for a Nix expression to create haskell-language-server binaries, seehttps://github.com/haskell/haskell-language-server/issues/122
```
Comment (jneira) [inline README.md:715]:
```suggestion
If you are looking for a Nix expression to create haskell-language-server binaries, see https://github.com/haskell/haskell-language-server/issues/122
```

**Comment by jneira** [comment]
I think this one would close https://github.com/haskell/haskell-language-server/issues/122, as we would have an "official" way to install using nix.

File: README.md (hunk 1 above)
Comment (jneira) [reply README.md:715]: could we include the nix expression in the readme, to being able to close the issue?

Comment (pepeiborra) [reply README.md:715]: But it's not a solved problem, and if it is I'm not in a position to state the solution, hence referring to an open issue.

**Review by Sir4ur0n — APPROVED** [review]
Thank you, this looks cleaner (though I'm no Nix expert)!

I have tested (with a 25 min Nix build :'( ) and this seems to work with Stack

File: nix/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{ sources ? import ./sources.nix }:
```
[the remark this answers is not in the document]
Comment (pepeiborra) [reply nix/default.nix:1]: You are welcome to do this, please send a PR!

File: README.md [hunk 3]
```diff
@@ -701,6 +701,19 @@ $ cd install
...
 ```
 
+#### Using Nix
+
+The instructions below show how to set up a Cachix binary cache and open a nix shell for local development.
+
+```shell
+$ cachix use haskell-language-server
+$ nix-shell
+$ cabal update
```
[the remark this answers is not in the document]
Comment (pepeiborra) [reply README.md:711]: This Nix script is specifically targeted at using with `cabal build`

File: nix/sources.json [hunk 4]
```diff
@@ -12,15 +24,15 @@
         "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
     },
     "nixpkgs": {
-        "branch": "nixpkgs-unstable",
+        "branch": "haskell-updates",
         "description": "A read-only mirror of NixOS/nixpkgs tracking the released channels. Send issues and PRs to",
```
[the remark this answers is not in the document]
Comment (pepeiborra) [reply nix/sources.json:28]: Done!

**Comment by pepeiborra** [comment]
> Thank you, this looks cleaner (though I'm no Nix expert)!
> 
> I have tested (with a 25 min Nix build :'( ) and this seems to work with Stack

Did you try to setup the Cachix binary cache? It should have downloaded all the artefacts for you (and stack will happily ignore them, but at least you don't have to wait for Nix)

**Pull request merged**